src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityQuery.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace D2L.CodeStyle.Analyzers.Immutability {
+	internal readonly struct ImmutabilityQuery {

```

Comment:
If we start caring about "dat perf" more, this might be a decent candidate to be a `ref struct`